docs/semgrep-secrets/historical-scanning.md

```diff
@@ -0,0 +1,114 @@
+    ```
+    semgrep ci --historical-secrets
+    ```
+1. Depending on your CI provider, you may have to perform additional steps to enable the job to run manually. For example, GitHub Actions requires the `workflow_dispatch` event to be added to your CI job.
+
+
+### Run a local test scan
+
+You can run a historical scan locally without sending the scan results to Semgrep Cloud Platform.
+
+You can run this test to determine the time it takes for Semgrep Secrets to run on your repository's git commit history.
+
+To run a test scan, enter the following command:
+
+```bash
+semgrep scan --historical-secrets
+```
+
+The historical scan results appear in the **Secrets Historical Scan** section:
+
+![Historical scan section in the CLI](/img/historical-scans-cli.png#md-width)
+
+```
+## View or hide historical findings
+
+![Historical secrets in Semgrep Cloud Platform](/img/historical-secrets-scp.png)
+*Figure.* Historical findings in Semgrep Cloud Platform.
+
+1. Sign in to [<i class="fas fa-external-link fa-xs"></i> Semgrep Cloud Platform](https://semgrep.dev/login).
+1. Click **<i class="fa-solid fa-key"></i> Secrets**. Historical findings are identified by an **<i class="fa-solid fa-hourglass-half"></i>** icon.
+1. Click **<i class="fa-solid fa-hourglass-half"></i> Hide historical** to toggle the display of historical findings.
+
+## How it works
+
+The following sections discuss historical scan limitations, how findings are triaged, and other details that may affect how you implement historical scans in your security program.
+
+### Limitations
+
+- Historical scanning can slow down scan times. Depending on the size of your repository history, it can take as quickly as under 5 minutes to more than 60 minutes for extreme cases.
+- Within SCP, historical scan findings are not automatically marked as **Fixed**. Findings can only exist in two states: `Open` or `Ignored`. Because Semgrep scans do not automatically detect historical findings as fixed, you must manually rotate and triage the secret as `Ignored`.
+- A finding can show up twice in the CLI with historical scanning enabled: the HEAD commit in the regular Secrets scan and another commit in the historical scan.
+    - If findings are sent to SCP, they are deduplicated and appear as a **regular finding**, not a historical finding.
+
+### Size of commit history
+
+- Semgrep Secrets scans up to **5 GiB** of uncompressed blobs. This ranges from around **10,000 to 50,000** previous commits depending on the average size of the commit.
+- For repositories with more than 5 GiB of history, Semgrep Secrets is still able to complete the scan, but the scan scope will not cover the older commits beyond 5 GiB.
+- The size of the commit history affects the speed of the scan. Larger repositories take longer to complete.
+- Semgrep Secrets scans the whole commit history every time it is run. This guarantees that your git history is also scanned using the **latest Secrets rules**.
```

Comment:
same full scan comment

Reply:
Nice catch! This + all other instances are fixed.